### DISCUSSION TITLE
[v3] useQueries : No type inference with two query fns of different return value types.

### DISCUSSION THREAD

**Discussion by bricejar** [TanStack/query#1319] [Ideas]
**Describe the bug**

When using **useQueries** with two queries of different return values, its return type defaults to  QueryObserverResult<unknown, unknown>[] .

**To Reproduce**

Check the function **useQueriesTypeIssue** in  https://codesandbox.io/s/restless-dust-ifoc9?file=/src/App.tsx .

**Expected behavior**
The return type of useQueries should be based on the type of the query functions passed in arguments.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome, Firefox
 - Version: Last

**Additional context**
Was asked to file a bug after a discussion in the official Discord.

**Comment by TkDodo** [comment]
I _also_ got no type inference when mapping over an array and using the same queryFn, so I think it's not specific to two different query functions.

**Comment by tannerlinsley** [comment]
This is a known drawback with the current types approach, but I plan on revisiting it post v3 launch.
